pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -60,6 +60,7 @@ func ipSecNewState() *netlink.XfrmState {
 		Mode:  netlink.XFRM_MODE_TUNNEL,
 		Proto: netlink.XFRM_PROTO_ESP,
 		ESN:   false,
+		Ifid:  1,
```

Comment:
I suspect the test are failing because this change is incorrect. In the bugtool reports, I see:
```
XfrmOutPolError         	32
```
and 
```
XfrmOutPolError         	8
```

Reply:
from my understanding, the kernel commit requires to set an Interface ID different from 0 for the state. The state is used to create a dummy interface:
https://github.com/cilium/cilium/blob/8e25d64386426ea97fa7b3a35f77d074ef4a7546/pkg/datapath/linux/ipsec/probe_linux.go#L60

Which is called when IPSec and tunneling are enabled:
https://github.com/cilium/cilium/blob/47fe2540e057917184abe54535f59764e1eb4f67/daemon/cmd/daemon_main.go#L1413-L1417

Reply:
> from my understanding, the kernel commit requires to set an Interface ID different from 0 for the state.

The kernel commit modifies functions `xfrmi_changelink` and `xfrmi_newlink`. AFAICS, these functions are only called when creating and changing attributes of xfrm interfaces.

> That's a good question since the probe should not interfere with the runtime.

The function you modified, `ipSecNewState`, is also called to create the actual xfrm states.

Reply:
This section looks interesting too: https://github.com/cilium/cilium/blob/8e25d64386426ea97fa7b3a35f77d074ef4a7546/pkg/datapath/linux/ipsec/ipsec_linux.go#L165-L169

If we check, we have the following part: 

https://github.com/cilium/cilium/blob/8e25d64386426ea97fa7b3a35f77d074ef4a7546/pkg/datapath/linux/ipsec/ipsec_linux.go#L156-L159

And `ipSecNewPolicy()` might keep the default `Ifid` to 0 instead of 1 and causing a mismatch which leads to the failure ? 
https://github.com/cilium/cilium/blob/8e25d64386426ea97fa7b3a35f77d074ef4a7546/pkg/datapath/linux/ipsec/ipsec_linux.go#L68

Reply:
Ah, yes. That could explain the errors and packet drops.

Reply:
@pchaigno it looks better for the "ConformanceKind1.19" check. :metal:

Reply:
Nice, but that doesn't explain why this change is needed. Did you check if it solves the issue you were initially hitting?

Reply:
To me it's explained with the elements mentioned in: https://github.com/cilium/cilium/pull/18789#discussion_r805711553.

> Did you check if it solves the issue you were initially hitting?

Currently adding a testcase to Flatcar test framework to reproduce the issue and I'll give a try with this changes to see if it solves the issue.With this minimal repro (https://gist.github.com/tormath1/bf3af973de9a4232698cc42199496496) I've been able to run the probe without the original issue.

Reply:
I'll have a look tomorrow regarding xfrm states and policies but I won't be able to compare with previous state since cilium pods was not even able to start correctly since the probing was failing. (see: https://github.com/flatcar-linux/Flatcar/issues/626). 
Thanks for kicking off the tests :)